modules/iam-assumable-role-with-oidc/main.tf

```diff
@@ -62,6 +62,28 @@ data "aws_iam_policy_document" "assume_role_with_oidc" {
   }
 }
 
+data "aws_iam_policy_document" "assume_self" {
```

Comment:
instead of a new document data source and then merging documents, we can use a dynamic statement block instead like

```hcl
  dynamic "statement" {
    for_each = var.explicit_permission_to_assume_self ? [1] : []
    
    content {
      ...
    }
  }
```